src/Microsoft.AspNetCore.OData/ODataController.cs

```diff
@@ -51,5 +52,108 @@ protected virtual UpdatedODataResult<TEntity> Updated<TEntity>(TEntity entity)
+        /// Creates a <see cref="StatusCodeResult"/> that when executed will produce a Bad Request (400) response.
+        /// </summary>
+        /// <param name="message">Error Message</param>
+        /// <returns>A <see cref="BadRequestODataResult"/> with the specified values.</returns>
+        protected virtual BadRequestODataResult BadRequest(string message)
+        {
+            return new BadRequestODataResult(message);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="StatusCodeResult"/> that when executed will produce a Bad Request (400) response.
+        /// </summary>
+        /// <param name="odataError">Parameter of type <see cref="ODataError"/>.</param>
+        /// <returns>A <see cref="BadRequestODataResult"/> with the specified values.</returns>
+        protected virtual BadRequestODataResult BadRequest(ODataError odataError)
+        {
+            return new BadRequestODataResult(odataError);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="StatusCodeResult"/> that when executed will produce a Not Found (404) response.
+        /// </summary>
+        /// <param name="message">Error Message</param>
+        /// <returns>A <see cref="NotFoundODataResult"/> with the specified values.</returns>
+        protected virtual NotFoundODataResult NotFound(string message)
+        {
+            return new NotFoundODataResult(message);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="StatusCodeResult"/> that when executed will produce a Not Found (404) response.
+        /// </summary>
+        /// <param name="odataError">Parameter of type <see cref="ODataError"/>.</param>
+        /// <returns>A <see cref="NotFoundODataResult"/> with the specified values.</returns>
+        protected virtual NotFoundODataResult NotFound(ODataError odataError)
+        {
+            return new NotFoundODataResult(odataError);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="StatusCodeResult"/> that when executed will produce a Unauthorized (401) response.
+        /// </summary>
+        /// <param name="message">Error Message</param>
+        /// <returns>A <see cref="UnauthorizedODataResult"/> with the specified values.</returns>
+        protected virtual UnauthorizedODataResult Unauthorized(string message)
+        {
+            return new UnauthorizedODataResult(message);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="StatusCodeResult"/> that when executed will produce a Unauthorized (401) response.
+        /// </summary>
+        /// <param name="odataError">Parameter of type <see cref="ODataError"/>.</param>
+        /// <returns>A <see cref="UnauthorizedODataResult"/> with the specified values.</returns>
+        protected virtual UnauthorizedODataResult Unauthorized(ODataError odataError)
+        {
+            return new UnauthorizedODataResult(odataError);
+        }
+
+#if NETCOREAPP3_1 || NETCOREAPP2_1 || NETSTANDARD2_1
```

Comment:
Could you add a comment on why this necessary and/or how it's different for .netframework?

Reply:
Updated based on Mike's suggestion.
Added a comment #Resolved